test/ServerRequestFactoryTest.php

```diff
@@ -62,6 +62,7 @@ public function testMarshalsExpectedHeadersFromServerArray()
             'HTTP_CONTENT_TYPE' => 'application/json',
             'HTTP_ACCEPT' => 'application/json',
             'HTTP_X_FOO_BAR' => 'FOOBAR',
+            'HTTP__1' => '-1',
```

Comment:
Based on RFC 7230, this should be interpreted as the following header:

```http
-1: -1
```

since `-` and numeric characters are all valid for header names. As such, this test is not correct.

I just added the above to the current test suite, and then updated the `$expected` value to include a `'-1' => '-1'` case, and it passes as expected.